### PR TITLE
Use decimal for a single decimal place fix

### PR DESCRIPTION
**Pull request by AbdiTolesa**
Reflects changes in https://github.com/Strategy11/formidable-payments/pull/81 which fixes https://github.com/Strategy11/formidable-pro/issues/4908

Quoting the original PR description:
Related ticket https://secure.helpscout.net/conversation/2493586368/188113/

> The payment amount is not accurate. It is setting a price of 1.115,00 € when it should be 111,50.

It would convert `111.50` to `111,50` but wasn't converting `111.5` to `111,5`, which causes the numbers to be off 10x more than expected.

I don't see harm in converting numbers with a single decimal place as well. I added a unit test to cover 3 "decimal places", which shouldn't be interpreted as a decimal.

## Summary by CodeRabbit

- **New Features**
	- Enhanced handling of decimal amounts in transactions to improve accuracy in financial calculations.
- **Tests**
	- Added new tests to ensure the reliability of decimal amount processing in transactions.

### CODE REVIEW

**Comment by AbdiTolesa** [comment]
@garretlaxton I haven't tested this since my Stripe account is blocked now but I believe it should work for you.

**Review by garretlaxton — APPROVED** [review]
This looks great!

File: stripe/controllers/FrmTransLiteActionsController.php [hunk 1]
```diff
@@ -275,8 +275,14 @@ private static function maybe_use_decimal( &$amount, $currency ) {
...
 
-		$amount_parts     = explode( '.', $amount );
-		$used_for_decimal = count( $amount_parts ) === 2 && strlen( $amount_parts[1] ) === 2;
+		$amount_parts = explode( '.', $amount );
+		if ( 2 !== count( $amount_parts ) ) {
+			return;
+		}
+
+		$strlen           = strlen( $amount_parts[1] );
+		$used_for_decimal = $strlen >= 1 && $strlen <= 2;
```
Comment (truongwp) [inline stripe/controllers/FrmTransLiteActionsController.php:284]: Should this be `$strlen === 1 || `$strlen === 2`?

Comment (AbdiTolesa) [reply stripe/controllers/FrmTransLiteActionsController.php]: I think so, thanks!

**Review by truongwp — APPROVED** [review]
This looks good. Thanks @AbdiTolesa!

**Review by Crabcyborg — APPROVED** [review]
Thanks @AbdiTolesa @truongwp and @garretlaxton!

🚀

**Pull request merged**